src/librustc_expand/mbe/macro_rules.rs

```diff
@@ -61,23 +61,8 @@ crate fn annotate_err_with_kind(
 }
 
 /// Instead of e.g. `vec![a, b, c]` in a pattern context, suggest `[a, b, c]`.
-fn suggest_slice_pat(e: &mut DiagnosticBuilder<'_>, site_span: Span, parser: &Parser<'_>) {
-    let mut suggestion = None;
-    if let Ok(code) = parser.sess.source_map().span_to_snippet(site_span) {
-        if let Some(bang) = code.find('!') {
```

Comment:
This should still work, but it should check for `"!["`, not just the bang (preexisting bug).

Oh look at that, this is a suggestion where the diagnostic mentions `-Z macro-backtrace`.

So this is basically the `r#async` thing, I bet that this suggestion still works but only with `-Z macro-backtrace`.